src/mpi/coll/transports/gentran/gentran_types.h

```diff
@@ -113,6 +129,7 @@ typedef struct MPII_Genutil_sched_t {
     int completed_vtcs;
     int last_fence;
 
+    MPIR_Request *req;
```

Comment:
This addition is not described in the commit message. Some rationale would be helpful.

Reply:
It is used in this same commit, in function vtx_issue. This is to support ibcast error propagation (e.g. along broadcast tree). The error is saved in req->u.nbc.errflag. so a pointer to the MPI_Request is added here. If I remember correctly, the MPIDU_Sched also store MPI_Request for the same reason for ibcast error checking.

Reply:
The next commit in the series adds error checking to ibcast. From what you are saying, this change belongs to that commit, not this one.

Reply:
Adding "req" to scheduler is part of that commit because the new vertex type added (for error checking) uses the "req" in the same commit. Moving req to later commit breaks it. I think even the commit to add schede_irecv_status technically speaking is also part of the error checking commit group.

Reply:
OK, that helps my understanding. It's sloppy commit boundaries like this that make it hard to review PRs, which results in things dragging out for a long time.

I'm not saying you need to go and fix the history this time. But please be aware for future PRs.

Reply:
Thanks for the comments. I understand reviewing this PR is challenging. Hopefully it is just this one. This PR is a collection of commits with pretty old history, which is also complicated by several rounds of code refactorization. 

Could you please also review https://github.com/pmodels/mpich/pull/5506 which was split from this PR.